src/Razor/src/Microsoft.CodeAnalysis.Remote.Razor/DocumentSymbols/RemoteDocumentSymbolService.cs

```diff
@@ -43,9 +44,47 @@ protected override IRemoteDocumentSymbolService CreateService(in ServiceArgs arg
             supportsVSExtensions: _clientCapabilitiesService.ClientCapabilities.SupportsVisualStudioExtensions,
             cancellationToken).ConfigureAwait(false);
 
+        // Roslyn uses an internal "RoslynDocumentSymbol" type, which throws when serialized after we've mapped it, so we have to
+        // convert things back to DocumentSymbol. We only need to do the first level though, as our remapping will take care of
+        // the children.
+        if (csharpSymbols.TryGetFirst(out var roslynDocumentSymbols))
+        {
+            csharpSymbols = ConvertDocumentSymbols(roslynDocumentSymbols);
+        }
+
         var codeDocument = await context.GetCodeDocumentAsync(cancellationToken).ConfigureAwait(false);
         var csharpDocument = codeDocument.GetCSharpDocument();
 
         return _documentSymbolService.GetDocumentSymbols(context.Uri, csharpDocument, csharpSymbols);
     }
+
+    [return: NotNullIfNotNull(nameof(roslynDocumentSymbols))]
```

Comment:
I don't think `DocumentSymbol[]?` is right for the signature. `TryGetFirst` on `SumType<DocumentSymbol[], SymbolInformation[]>` should guarantee a non-null value I think